File: ePubAnalyzer.Shared/Entities/Book.cs
namespace ePubAnalyzer.Shared.Entities
{
	public class Book
	{
		public int? BookID {get;set;}
		public string ReadStatus {get;set;}
		public string ReadRemark {get;set;}
		public string Identifier {get;set;}
		public string Title {get;set;}
		public string Author {get;set;}
		public string Category {get;set;}
		public string Subject {get;set;}
		public string Description {get;set;}
		public string Language {get;set;}
		public string Folder {get;set;}
		public string FileName {get;set;}
		public string Medium {get;set;}
		//Available, or lend to somebody
		public string Status {get;set;}
		//Remark applicable when lend to somebody
		public string StatusRemark {get;set;}
		public int? NrOfPages {get;set;}
		public override string ToString()
		{
			return $"{Author} - {Title} (identifier: {Identifier}, bookID: {BookID})";
		}
	}
}

File: Catalog.NG/src/book.ts
export interface IBook {
	Author?: string;
	BookID: number;
	Category?: string;
	Description?: string;
	FileName?: string;
	Folder?: string;
	Identifier?: string;
	Language?: string;
	Medium?: string;
	ReadRemark?: string;
	ReadStatus?: string;
	Status?: string;
	StatusRemark?: string;
	Subject?: string;
	Title?: string;
	NrOfPages?: number;
}

// Default update model (exposint bookId)
export interface IBookUpdateModel {
    BookID?: number;
}

// Model for saving book details (like the regular book interface)
export interface IBookDetailUpdateModel extends IBookUpdateModel {
	
    Identifier?: string;
    Title?: string;
    Author?: string;
    Description?: string;
	Medium?: string;
    NrOfPages?: number;
}

// Update the read status badge (no desc)
export interface IBookReadBadgeUpdateModel extends IBookUpdateModel {  
    ReadStatus?: string;
}

// Full read status update model (including comment)
export interface IBookReadStatusUpdateModel extends IBookReadBadgeUpdateModel {
	ReadRemark?: string;
}

// Update the availability status
export interface IBookAvailabilityStatusUpdateModel extends IBookUpdateModel {
    Status?: string;
    StatusRemark?: string;
}